kubernetes/RemoteResource/resource.yaml

```diff
@@ -233,13 +233,34 @@ items:
                             oneOf:
                               - required: [url]
                               - required: [uri]
+                              - required: [github]
```

Comment:
do we need to tightly couple this with `github`? or is it possible to do a more general grouping like `git` that way we leave ourselves open for something like gitlab in the future? Im not sure how this is being implemented, so it might not be possible. what do you think @kdai7 ?

Reply:
Since the github and gitlab request requirements are different I was going to separate them, but I could still group the separate requirements under git. I'll just wait on the pr until I'm done with gitlab then.

Reply:
how are they different? are these things we want to hardcode or something the user should be providing?